### PR TITLE
Bug 1755757: store: add kube_node_role metric

### PR DESCRIPTION
**Pull request by brancz**
Backport of #16 for 4.1.

@LiliC @paulfantom @LiliC

### CODE REVIEW

**Comment by paulfantom** [comment]
@brancz we need BZ for it

**Comment by brancz** [comment]
I verified this works on a cluster bot cluster.

**Comment by LiliC** [comment]
@brancz I duplicated the original bugzilla and target it to the 4.1 release, can you refernce that bugzilla please, I can't seem to edit the title 🤷‍♀  https://bugzilla.redhat.com/show_bug.cgi?id=1750283

**Comment by bparees** [comment]
@brancz @LiliC the 4.2 BZ (https://bugzilla.redhat.com/show_bug.cgi?id=1749451) is still in assigned, we'll need to see that get fixed/verified before this PR can merge into a z-stream

**Comment by LiliC** [comment]
@bparees thanks, this is now ready to be merged.

**Comment by LiliC** [comment]
Seems like the wrong bugzilla is linked, should be https://bugzilla.redhat.com/show_bug.cgi?id=1755757

**Comment by bparees** [comment]
> @bparees thanks, this is now ready to be merged.

Thanks, I believe @sdodson is patch manager through EOD today so he should see this in his query and tag it today assuming he deems it ready (it looks ready to me).

**Comment by sdodson** [comment]
There's other work involved to resolve the bug in question that won't be queued before the merge window closes, therefore we'll defer this until the next 4.1.z merge window.

**Comment by LiliC** [comment]
@sdodson I believe we said we can merge today? I changed the bugzilla to medium priority as per the prior discussion.

**Comment by sdodson** [comment]
@mrunalp starts his term as patch manager today. I'll sync up with him when he's online.

File: pkg/collectors/node_test.go [hunk 1]
```diff
@@ -127,9 +127,10 @@ func TestNodeCollector(t *testing.T) {
 				},
 			},
 			Want: `
-        kube_node_created{node="127.0.0.1"} 1.5e+09
+		kube_node_created{node="127.0.0.1"} 1.5e+09
         kube_node_info{container_runtime_version="rkt",kernel_version="kernel",kubelet_version="kubelet",kubeproxy_version="kubeproxy",node="127.0.0.1",os_image="osimage",provider_id="provider://i-randomidentifier"} 1
-        kube_node_labels{label_type="master",node="127.0.0.1"} 1
+		kube_node_labels{label_node_role_kubernetes_io_master="",node="127.0.0.1"} 1
+		kube_node_role{node="127.0.0.1",role="master"} 1
```
Comment (mrunalp) [inline pkg/collectors/node_test.go]: nit: Are we missing a gofmt run here?

Comment (LiliC) [reply pkg/collectors/node_test.go]: This is to sync with the upstream repo, if we were to make the change here we would diverge with the upstream branch.

Comment (mrunalp) [reply pkg/collectors/node_test.go]: okay!

**Pull request merged**